src/GitHubClient.php

```diff
@@ -100,7 +102,12 @@ public function __construct(Composer $composer, IOInterface $io)
     {
         $this->composer = $composer;
         $this->io = $io;
-        $this->rfs = Factory::createRemoteFilesystem($io, $composer->getConfig());
+
+        if (version_compare('2', $composer::getVersion()) < 0) {
```

Comment:
Can we use feature detection instead?

```suggestion
        if (class_exists('Composer\Util\RemoteFilesystem')) {
```

Reply:
Thanks for the review. 

I initially thought to use a `class_exists()` call too. I ended up using a `version_compare` like here because Composer 2 is just starting to separate its file system and HTTP fetcher components, and a `version_compare` would make it more readable and clear the intent that we branch our code depending on the version.

Do you think version comparing is fragile in a way, or be an anti-pattern? I can make amends if you think we should continue with feature detection. Thanks.

Reply:
Well, `class_exists()` should be faster. I don't know if that matters here though.

> a `version_compare` would make it more readable and clear the intent that we branch our code depending on the version.

Well, I'd say we branch our code depending on the availability of a feature. 🤓 

> Do you think version comparing is fragile in a way, or be an anti-pattern?

Personally, I'd consider it an anti-pattern that I avoid if I can. Checking for the availability of a class that I'm about to use feels like the more robust thing to do.

But I don't have a strong opinion here. I just thought, we should have discussed the options. 😃

Reply:
Great, thank you. I just pushed a new commit with these changes. Thank you!